packages/shaders-react/src/shader-mount.tsx

```diff
@@ -81,59 +80,69 @@ const processUniforms = (uniforms: ShaderMountUniformsReact): Promise<ShaderMoun
  * A React component that mounts a shader and updates its uniforms as the component's props change
  * If you pass a string as a uniform value, it will be assumed to be a URL and attempted to be loaded as an image
  */
-export const ShaderMount: React.FC<ShaderMountProps> = ({
-  ref,
-  shaderMountRef: externalShaderMountRef,
-  fragmentShader,
-  style,
-  uniforms = {},
-  webGlContextAttributes,
-  speed = 1,
-  frame = 0,
-}) => {
-  const canvasRef = ref ?? useRef<HTMLCanvasElement>(null);
-  const shaderMountRef = externalShaderMountRef ?? useRef<ShaderMountVanilla | null>(null);
```

Comment:
- Hooks shouldn't be called conditionally
- `canvasRef` would break work with React 19 callback refs